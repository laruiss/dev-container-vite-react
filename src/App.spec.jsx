import React from 'react'
import { render } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import App from './App.jsx'

describe('App', () => {
  it('should mount App component', () => {
    const component = render(
      <App />,
    )
    const { asFragment } = component
    expect(asFragment).toMatchSnapshot()
  })

  it('should increment count', () => {
    // Given
    const component = render(
      <App />,
    )
    const button = component.getByText(/count is: 0/i)

    // When
    userEvent.click(button)

    // Then
    expect(button.innerHTML).toContain('count is: 1')
  })
})
