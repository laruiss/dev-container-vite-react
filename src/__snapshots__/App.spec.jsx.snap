// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should mount App component 1`] = `[Function]`;
